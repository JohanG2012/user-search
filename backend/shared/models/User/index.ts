import { Document, model, Schema } from 'mongoose';

interface IName extends Document {
  title: string;
  first: string;
  last: string;
}

interface IPicture extends Document {
  large: string;
  medium: string;
  thumbnail: string;
}

interface IUser extends Document {
  name: IName;
  picture: IPicture;
  permission: boolean;
}

const userSchema = new Schema(
  {
    name: {
      title: { type: String, required: true },
      first: { type: String, required: true },
      last: { type: String, required: true },
    },
    picture: {
      large: { type: String, required: true },
      medium: { type: String, required: true },
      thumbnail: { type: String, required: true },
    },
    permission: { type: Boolean },
  },
  {
    versionKey: false,
  },
);

export default model<IUser>('User', userSchema);
