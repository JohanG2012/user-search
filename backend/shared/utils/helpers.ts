export const stripObjProps = (inputArr: object[], stripArr: string[]): object[] => {
  return inputArr.map(obj => {
    const newObj = { ...obj };
    stripArr.forEach(str => {
      delete newObj[str];
    });
    return newObj;
  });
};
