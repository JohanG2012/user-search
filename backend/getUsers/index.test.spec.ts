/* tslint:disable no-var-requires */
const mongoose = require('mongoose');
/* tslint:enable no-var-requires */
import { MongoMemoryServer } from 'mongodb-memory-server';
import httpFunction from './index';
import User from '../shared/models/User';
import * as testData from '../data/testData.json'; // 100 user documents
import { stripObjProps } from '../shared/utils/helpers';
mongoose.Promise = global.Promise;

let mongoServer;

beforeAll(async () => {
  mongoServer = new MongoMemoryServer();
  const mongoUri = await mongoServer.getConnectionString();
  await mongoose.connect(mongoUri, err => {
    if (err) {
      console.error(err);
    }
  });

  // Promise fix for broken insertMany
  await new Promise((resolve, reject) => {
    User.insertMany(testData, err => {
      if (err) {
        reject(err);
      }
      resolve();
    });
  });
});

afterAll(async () => {
  await mongoose.disconnect();
  await mongoServer.stop();
});

describe('GET /api/v1/users', () => {
  let context: any = {};
  let req: any = {};
  beforeEach(() => {
    context = {
      res: { body: {} },
      done: jest.fn(),
      log: jest.fn(),
    };

    req = {
      params: {},
      query: {},
    };
  });

  describe('without querystrings', () => {
    beforeEach(async () => {
      await httpFunction(context, req);
    });
    it('should match snapshot', async () => {
      // strip _id & next to avoid snapshot collision.
      const parsed = JSON.parse(context.res.body);
      const [stripedBody]: any = stripObjProps([parsed], ['next']);
      const striped = stripObjProps(parsed.data, ['_id']);
      stripedBody.data = striped;
      context.res.body = JSON.stringify(stripedBody);

      expect(context.done).toBeCalledTimes(1);
      expect(context).toMatchSnapshot();
    });
    describe('few fallbacks in case bad snapshot is saved', () => {
      it('should return 200 status code', () => {
        expect(context.res.status).toBe(200);
      });
      it('should return list of first 50 users', () => {
        expect(JSON.parse(context.res.body).data.length).toBe(50);
      });
    });
  });
  describe('with all querystrings', () => {
    beforeEach(async () => {
      req.query = { limit: 10, 'name.title': 'mr', fields: 'name.title,picture.thumbnail' };
      await httpFunction(context, req);
    });
    describe('with limit', () => {
      it('should return a list of 10 users', () => {
        expect(JSON.parse(context.res.body).data.length).toBe(10);
      });
    });
    describe('with filter', () => {
      it('should return users with the title mr', () => {
        const result = JSON.parse(context.res.body).data;
        result.forEach(item => {
          expect(item.name.title).toBe('mr');
        });
      });
    });
    describe('with partial', () => {
      it('should not return picture.large and picture.medium', () => {
        const result = JSON.parse(context.res.body).data;
        result.forEach(item => {
          expect(item.picture.large).toBeUndefined();
          expect(item.picture.medium).toBeUndefined();
        });
      });
    });
    it('should match snapshot', () => {
      expect(context).toMatchSnapshot();
    });
  });
  describe('validation', () => {
    describe('with to high limit', () => {
      beforeEach(async () => {
        req.query.limit = 60;
        await httpFunction(context, req);
      });
      it('should return 400 status code', async () => {
        expect(context.res.status).toBe(400);
      });
      it('should match snapshot', () => {
        expect(context).toMatchSnapshot();
        expect(context.done).toHaveBeenCalledTimes(1);
      });
    });
    describe('with invalid next cursor', () => {
      beforeEach(async () => {
        req.query.next = 'fake cursor';
        await httpFunction(context, req);
      });
      it('should return 400 status code', async () => {
        expect(context.res.status).toBe(400);
      });
      it('should match snapshot', () => {
        expect(context).toMatchSnapshot();
        expect(context.done).toHaveBeenCalledTimes(1);
      });
    });
  });
  describe("search for 'justine faure'", () => {
    it('should match snapshot', async () => {
      req.query.search = 'justine faure';
      await httpFunction(context, req);

      // strip _id to avoid snapshot collision.
      const parsed = JSON.parse(context.res.body);
      const striped = stripObjProps(parsed.data, ['_id']);
      parsed.data = striped;
      context.res.body = JSON.stringify(parsed);

      expect(context).toMatchSnapshot();
      expect(context.done).toHaveBeenCalledTimes(1);
    });
  });
  describe('with limit & pagination', () => {
    it('should respond with page 2', async () => {
      req.query.limit = 3;
      await httpFunction(context, req);
      const ref = JSON.parse(context.res.body).data;
      req.query.limit = 2;
      await httpFunction(context, req);
      let parsed = JSON.parse(context.res.body);
      expect(parsed.data.length).toBe(2);
      expect(parsed.data[1]).toMatchObject(ref[1]);
      const next = parsed.next;
      req.query.next = next;
      await httpFunction(context, req);
      parsed = JSON.parse(context.res.body);
      expect(parsed.data[0]).toMatchObject(ref[2]);
    });
  });
  describe('trigger server error - close down database connection', () => {
    beforeEach(async () => {
      mongoose.disconnect();
      mongoServer.stop();
      await httpFunction(context, req);
    });
    it('should respond with 500 error', async () => {
      expect(context.res.status).toBe(500);
    });
    it('should match snapshot', async () => {
      expect(context).toMatchSnapshot();
      expect(context.done).toHaveBeenCalledTimes(1);
    });
  });
});
