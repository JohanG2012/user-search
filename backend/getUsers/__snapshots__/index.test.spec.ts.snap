// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /api/v1/users search for 'justine faure' should match snapshot 1`] = `
Object {
  "done": [MockFunction] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "log": [MockFunction],
  "res": Object {
    "body": "{\\"status\\":\\"success\\",\\"data\\":[{\\"name\\":{\\"title\\":\\"madame\\",\\"first\\":\\"justine\\",\\"last\\":\\"faure\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/79.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/79.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/79.jpg\\"}}]}",
    "headers": Object {
      "Content-Type": "applicaiton/json",
    },
    "status": 200,
  },
}
`;

exports[`GET /api/v1/users trigger server error - close down database connection should match snapshot 1`] = `
Object {
  "done": [MockFunction] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "log": [MockFunction] {
    "calls": Array [
      Array [
        "Topology was destroyed",
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "res": Object {
    "body": "{\\"status\\":\\"error\\",\\"message\\":\\"Oops! Something unknown happend. Please open an issues with intstructions on how to reproduce this response.\\",\\"errorCode\\":2003}",
    "headers": Object {
      "Content-Type": "application/json",
    },
    "status": 500,
  },
}
`;

exports[`GET /api/v1/users validation with invalid next cursor should match snapshot 1`] = `
Object {
  "done": [MockFunction] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "log": [MockFunction],
  "res": Object {
    "body": "{\\"status\\":\\"failed\\",\\"field\\":\\"next\\",\\"message\\":\\"Not a valid cursor\\",\\"errorCode\\":2002}",
    "headers": Object {
      "Content-Type": "applicaiton/json",
    },
    "status": 400,
  },
}
`;

exports[`GET /api/v1/users validation with to high limit should match snapshot 1`] = `
Object {
  "done": [MockFunction] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "log": [MockFunction],
  "res": Object {
    "body": "{\\"status\\":\\"failed\\",\\"field\\":\\"limit\\",\\"message\\":\\"\\\\\\"limit\\\\\\" must be less than or equal to 50\\",\\"errorCode\\":2002}",
    "headers": Object {
      "Content-Type": "applicaiton/json",
    },
    "status": 400,
  },
}
`;

exports[`GET /api/v1/users with all querystrings should match snapshot 1`] = `
Object {
  "done": [MockFunction] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "log": [MockFunction],
  "res": Object {
    "body": "{\\"status\\":\\"success\\",\\"data\\":[{\\"name\\":{\\"title\\":\\"mr\\"},\\"picture\\":{\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/55.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\"},\\"picture\\":{\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/25.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\"},\\"picture\\":{\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/18.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\"},\\"picture\\":{\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/61.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\"},\\"picture\\":{\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/22.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\"},\\"picture\\":{\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/9.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\"},\\"picture\\":{\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/33.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\"},\\"picture\\":{\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/29.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\"},\\"picture\\":{\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/6.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\"},\\"picture\\":{\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/80.jpg\\"}}]}",
    "headers": Object {
      "Content-Type": "applicaiton/json",
    },
    "status": 200,
  },
}
`;

exports[`GET /api/v1/users without querystrings should match snapshot 1`] = `
Object {
  "done": [MockFunction] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "log": [MockFunction],
  "res": Object {
    "body": "{\\"status\\":\\"success\\",\\"data\\":[{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"jerry\\",\\"last\\":\\"sullivan\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/55.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/55.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/55.jpg\\"}},{\\"name\\":{\\"title\\":\\"mademoiselle\\",\\"first\\":\\"mia\\",\\"last\\":\\"berger\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/61.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/61.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/61.jpg\\"}},{\\"name\\":{\\"title\\":\\"madame\\",\\"first\\":\\"justine\\",\\"last\\":\\"faure\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/79.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/79.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/79.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"javier\\",\\"last\\":\\"reyes\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/25.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/25.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/25.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"elias\\",\\"last\\":\\"leino\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/18.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/18.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/18.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"ray\\",\\"last\\":\\"torres\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/61.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/61.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/61.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"lohan\\",\\"last\\":\\"muller\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/22.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/22.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/22.jpg\\"}},{\\"name\\":{\\"title\\":\\"mrs\\",\\"first\\":\\"آیناز\\",\\"last\\":\\"علیزاده\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/83.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/83.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/83.jpg\\"}},{\\"name\\":{\\"title\\":\\"miss\\",\\"first\\":\\"sheila\\",\\"last\\":\\"austin\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/31.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/31.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/31.jpg\\"}},{\\"name\\":{\\"title\\":\\"monsieur\\",\\"first\\":\\"sasha\\",\\"last\\":\\"boyer\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/10.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/10.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/10.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"adam\\",\\"last\\":\\"davis\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/9.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/9.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/9.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"davi\\",\\"last\\":\\"koskamp\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/33.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/33.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/33.jpg\\"}},{\\"name\\":{\\"title\\":\\"mrs\\",\\"first\\":\\"isabella\\",\\"last\\":\\"lavigne\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/61.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/61.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/61.jpg\\"}},{\\"name\\":{\\"title\\":\\"ms\\",\\"first\\":\\"latife\\",\\"last\\":\\"akal\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/49.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/49.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/49.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"murat\\",\\"last\\":\\"uluhan\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/29.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/29.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/29.jpg\\"}},{\\"name\\":{\\"title\\":\\"miss\\",\\"first\\":\\"olivia\\",\\"last\\":\\"curtis\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/63.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/63.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/63.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"kasper\\",\\"last\\":\\"petersen\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/6.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/6.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/6.jpg\\"}},{\\"name\\":{\\"title\\":\\"monsieur\\",\\"first\\":\\"samuel\\",\\"last\\":\\"lucas\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/46.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/46.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/46.jpg\\"}},{\\"name\\":{\\"title\\":\\"monsieur\\",\\"first\\":\\"aubin\\",\\"last\\":\\"pierre\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/30.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/30.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/30.jpg\\"}},{\\"name\\":{\\"title\\":\\"madame\\",\\"first\\":\\"noélie\\",\\"last\\":\\"vidal\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/93.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/93.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/93.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"علی\\",\\"last\\":\\"زارعی\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/80.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/80.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/80.jpg\\"}},{\\"name\\":{\\"title\\":\\"mrs\\",\\"first\\":\\"edeolinda\\",\\"last\\":\\"ramos\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/81.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/81.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/81.jpg\\"}},{\\"name\\":{\\"title\\":\\"miss\\",\\"first\\":\\"kylie\\",\\"last\\":\\"garrett\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/23.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/23.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/23.jpg\\"}},{\\"name\\":{\\"title\\":\\"miss\\",\\"first\\":\\"freya\\",\\"last\\":\\"hughes\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/65.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/65.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/65.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"alco\\",\\"last\\":\\"stavenuiter\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/91.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/91.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/91.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"cyril\\",\\"last\\":\\"van de vliert\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/60.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/60.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/60.jpg\\"}},{\\"name\\":{\\"title\\":\\"ms\\",\\"first\\":\\"chloe\\",\\"last\\":\\"king\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/15.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/15.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/15.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"mathis\\",\\"last\\":\\"hubert\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/84.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/84.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/84.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"byron\\",\\"last\\":\\"owens\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/41.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/41.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/41.jpg\\"}},{\\"name\\":{\\"title\\":\\"mrs\\",\\"first\\":\\"jasmina\\",\\"last\\":\\"miller\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/24.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/24.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/24.jpg\\"}},{\\"name\\":{\\"title\\":\\"madame\\",\\"first\\":\\"norah\\",\\"last\\":\\"girard\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/89.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/89.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/89.jpg\\"}},{\\"name\\":{\\"title\\":\\"ms\\",\\"first\\":\\"greta\\",\\"last\\":\\"sauer\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/50.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/50.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/50.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"väinö\\",\\"last\\":\\"suomi\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/54.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/54.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/54.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"anderson\\",\\"last\\":\\"van der wind\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/24.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/24.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/24.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"علیرضا\\",\\"last\\":\\"احمدی\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/14.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/14.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/14.jpg\\"}},{\\"name\\":{\\"title\\":\\"monsieur\\",\\"first\\":\\"julian\\",\\"last\\":\\"meunier\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/5.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/5.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/5.jpg\\"}},{\\"name\\":{\\"title\\":\\"miss\\",\\"first\\":\\"lucie\\",\\"last\\":\\"picard\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/57.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/57.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/57.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"alfredo\\",\\"last\\":\\"da conceição\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/80.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/80.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/80.jpg\\"}},{\\"name\\":{\\"title\\":\\"ms\\",\\"first\\":\\"wiepke\\",\\"last\\":\\"wentink\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/45.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/45.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/45.jpg\\"}},{\\"name\\":{\\"title\\":\\"miss\\",\\"first\\":\\"تارا\\",\\"last\\":\\"صدر\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/49.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/49.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/49.jpg\\"}},{\\"name\\":{\\"title\\":\\"ms\\",\\"first\\":\\"annabelle\\",\\"last\\":\\"zhang\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/96.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/96.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/96.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"noah\\",\\"last\\":\\"teixeira\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/34.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/34.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/34.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"vilho\\",\\"last\\":\\"koskinen\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/65.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/65.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/65.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"german\\",\\"last\\":\\"bravo\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/83.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/83.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/83.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"tracy\\",\\"last\\":\\"ward\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/10.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/10.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/10.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"jeff\\",\\"last\\":\\"bailey\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/25.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/25.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/25.jpg\\"}},{\\"name\\":{\\"title\\":\\"miss\\",\\"first\\":\\"florence\\",\\"last\\":\\"miles\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/36.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/36.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/36.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"aaron\\",\\"last\\":\\"owens\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/19.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/19.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/19.jpg\\"}},{\\"name\\":{\\"title\\":\\"miss\\",\\"first\\":\\"eléonore\\",\\"last\\":\\"arnaud\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/women/80.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/women/80.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/women/80.jpg\\"}},{\\"name\\":{\\"title\\":\\"mr\\",\\"first\\":\\"jake\\",\\"last\\":\\"evans\\"},\\"picture\\":{\\"large\\":\\"https://randomuser.me/api/portraits/men/44.jpg\\",\\"medium\\":\\"https://randomuser.me/api/portraits/med/men/44.jpg\\",\\"thumbnail\\":\\"https://randomuser.me/api/portraits/thumb/men/44.jpg\\"}}]}",
    "headers": Object {
      "Content-Type": "applicaiton/json",
    },
    "status": 200,
  },
}
`;
